src/components/Select/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { ChangeEvent, Component } from 'react';
+import { SelectProps } from '../../types/component';
+
+class Select extends Component<SelectProps> {
+  shouldComponentUpdate(nextProps: SelectProps) {
+    return nextProps.options !== this.props.options;
+  }
+
+  handleChangeOption = (event: ChangeEvent<HTMLSelectElement>) => {
+    const target = event.target as HTMLSelectElement;
+    const result = {
+      [this.props.attributes.name]: target.value,
+    };
+    this.props.onChange(result);
+  };
+
+  createOptionElements = () => {
+    const { options } = this.props;
+
+    return options.map((option) => (
+      <option key={option} value={option}>
+        {option}
+      </option>
+    ));
+  };
+
+  render() {
+    const { attributes } = this.props;
+
+    return (
+      <label htmlFor={attributes.id}>
+        <select
+          name={attributes.name}
+          id={attributes.id}
+          className={attributes.className}
+          onChange={this.handleChangeOption}
+        >
+          {this.createOptionElements()}
+        </select>
+      </label>
+    );
```

Comment:
```suggestion
  render() {
    const { attributes } = this.props;

    return (
      <label htmlFor={attributes.id}>
        <select
          name={attributes.name}
          id={attributes.id}
          className={attributes.className}
          onChange={this.handleChangeOption}
        >
          {this.props.options.map((option) => (
            <option key={option} value={option}>
              {option}
            </option>
          ))}
        </select>
      </label>
    );
  }
```

너무 길지 않다면 jsx안에서 바로 쓰는게 일반적인 거 같아요!

Reply:
그렇군요!!

스텝 2에서 밑에서 알려주신 `HTMLAttributes<HTMLSelectElement>`를 사용하면서 하나하나 안 쓰고 `{...attributes}`로 `select`태그에 필요한 정보들을 넘겨줬습니다!